### PR TITLE
some additional fixes for issue #93: strip authentication even when all-...

### PR DESCRIPTION
**Pull request by ntfrnzn**
...headers is not set, allow lowercase header name, fix an offset bug, add go tests

### CODE REVIEW

**Comment by ntfrnzn** [comment]
I finally got around to running version 0.5 and realized that my changes to strip http basic authentication (issue #93) had a couple errors.  (a) miscalculated the offset for everything but my primitive test (b) required that all-headers be set (c) assumed that http header names were always capitalized.   Ouch, sorry about those.  I added some tests to the go test suite that catch these problems at least.

**Comment by tsg** [comment]
Hi, thanks for the Pull Request. Before merging this one, there's I'd like to ask you (I tried to email you but got a bounce from the address in your git config).

We have exciting plans for this project and we'd like to change the license for the next version of Packetbeat from GPLv2 to Apache 2, which we think will encourage the adoption even more. You can read the Apache 2 license [here](http://www.apache.org/licenses/LICENSE-2.0.html) or get a quick overview [here](https://tldrlegal.com/license/apache-license-2.0-%28apache-2.0%29).

You have contributed code to the project, so we'd like to ask for your approval before we do the change. If you agree with the license change, a simple reply here saying that you are fine with it would be enough. If you have any questions, let me know.

**Comment by ntfrnzn** [comment]
Yes, that's absolutely fine.  No problems at all with the Apache 2
license.

Now to fix my github profile :)

Nathan

**Comment by ruflin** [comment]
@ntfrnzn Quite a few changes happened on packetbeat since this pull request was opened. Can you merge in master to resolve the conflicts?

**Comment by ntfrnzn** [comment]
@ruflin It's been a while since I looked at this, yes.  Should be good now.   Also, signed the CLA,

File: protos/http/http.go [hunk 1]
```diff
@@ -957,25 +955,28 @@ func (http *Http) hideHeaders(m *HttpMessage, msg []byte) {
...
-			val_end_x := -1
-			if val_start_x != -1 {
-				val_end_x = bytes.Index(msg[m.headerOffset+val_start_x:m.bodyOffset], []byte("\r\n"))
-
-				if val_end_x < 0 || val_end_x > header_len {
-					val_end_x = header_len
+		auth_text := []byte("uthorization:")   // [aA]
+		if http.Strip_authorization {
+
+			authheader_start_x := m.headerOffset + bytes.Index(msg[m.headerOffset:m.bodyOffset], auth_text)
```
Comment (ruflin) [inline protos/http/http.go]: Can you use camelCase for the newly introduced variables? We are currently in the progress of refactoring all variable names. So all the new ones we introduce we already use the new naming scheme. Only change the ones that are local to your changes.

**Comment by ruflin** [comment]
@ntfrnzn Thanks for the update and singing the CLA. It currently still shows a merge conflict?

File: tests/test_0012_http_basicauth.py [hunk 2]
```diff
@@ -1,24 +1,43 @@
 from pbtests.packetbeat import TestCase
 import re
 
+"""
+Tests for removing base64-encoded authentication information
```
Comment (urso) [inline tests/test_0012_http_basicauth.py]: s/removing/blackening/

**Comment by urso** [comment]
got confused by "strip", as it's more like blakening, hehe.

Maybe add another test including Proxy-Authorization.

**Comment by ntfrnzn** [comment]
Thanks for the tip about Proxy-Authorization, that hadn't turned up in my work but it's important.  I made changes to the docs also -- I hope in the right place?  Sorry about the confusion between "authorization" [US] and "authentication".  In my opinion the HTTP header has it wrong in calling it "authorization" and not "authentication" but there's no fix for that now.

**Comment by urso** [comment]
Thanks, looks good.

**Pull request merged**